test/helpers/vscodefake/memento.ts

```diff
@@ -1,9 +1,29 @@
 import * as vscode from 'vscode';
 
 export class TestMemento implements vscode.Memento {
-  private storage: {[key: string]: any} = {};
+  private storage = new Map<string, any>();
 
-  public get<T>(key: string): T|undefined;
+  public get<T>(key: string): T | undefined;
+  public get<T>(key: string, defaultValue: T): T;
+  get<T>(key: string, defaultValue?: T): T | undefined {
+    const value = this.storage.get(key) as T | undefined;
+    if (value === undefined) {
+      return defaultValue;
+    }
+    return value;
+  }
+  public update(key: string, value: any): Thenable<void> {
+    this.storage.set(key, value);
+    return Promise.resolve();
+  }
+  public containsKey(key: string): boolean { return this.storage.hasOwnProperty(key); }
+  public clear() { this.storage.clear(); }
+}
+
+export class StatetMemento implements vscode.Memento {
```

Comment:
misspelled?  Should it be `StateMemento`?

Reply:
Do we need both this class and `TestMemento`?  Would the tests still work if we just kept one of them?